packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -924,7 +932,19 @@ export default class NewVm extends BaseComponent {
   _getRedirectionUrl = id =>
     this.state.state.multipleVms ? '/home' : `/vms/${id}`
 
-  _handleBootFirmware = value => this._setState({ hvmBootFirmware: value })
+  _handleBootFirmware = value => {
+    const isUefi = value === 'uefi'
+    this.setState(({ state }) => ({
+      state: {
+        ...state,
+        copyHostBiosStrings:
+          isUefi || this._hasBiosStrings()
+            ? !isUefi
+            : state.copyHostBiosStrings,
+        hvmBootFirmware: value,
+      },
+    }))
+  }
```

Comment:
It's usually better to passively handle these cases than to try to actively make all the necessary changes when something happens. So I suggest:
- When the HVM boot firmware is set by the user: `setState({ hvmBootFirmware })` (no change)
- When the "Copy BIOS strings" checkbox is checked/unchecked by the user: `toggleState('copyHostBiosStrings')` (as you did)
- The "Copy BIOS strings" `checked` value should be `value !== 'uefi' && (this._templateHasBiosStrings() || state.copyHostBiosStrings)`
- The "Copy BIOS strings" `disabled` value should be `value === 'uefi' || this._templateHasBiosStrings()` (as you did)
- When the form is initialized: `copyHostBiosStrings: !isEmpty(template.bios_strings)` (as you did)
- When the creation form is submitted: `copyHostBiosStrings: value !== 'uefi' && (this._templateHasBiosStrings() || state.copyHostBiosStrings)`